### DISCUSSION TITLE
Support backups volume mapped to an S3 bucket

### DISCUSSION THREAD

**Discussion by Cooke** [prodrigestivill/docker-postgres-backup-local#93] [Ideas]
Thanks for a great tool.

I'm using it with a volume mapped to an S3 bucket. Because of this the hard linking in the script fails. 

I could probably make a PR if its welcome to optionally use file copy instead of linking?

**Comment by prodrigestivill** [comment]
Hello @Cooke,

For using with S3 isn't it better to use the original image [schickling/postgres-backup-s3](https://hub.docker.com/r/schickling/postgres-backup-s3/)?

As it is implemented right now copying would make you use a lot of data usage for your S3, since as it is implemented it would have to upload 3 times the exact same file after each backup.

Maybe a better approach if you still want to use this docker image would be use [rclone](https://hub.docker.com/r/rclone/rclone) with the backups volume shared to upload the `daily` folder to your S3 (or `last` folder if I finally merge [feature/limit-last-backups](/prodrigestivill/docker-postgres-backup-local/tree/feature/limit-last-backups) branch), but don't upload all the files in the backups volume to avoid the issue stated before.

I hope this information helps in your deployment.

**Comment by petre-c** [comment]
Greetings, gentlemen 🎩,
Firstly, thank you @prodrigestivill for the excellent work with the tool.

[postgres-backup-s3](https://github.com/eeshugerman/postgres-backup-s3/issues/40) doesn't support retention policies that are available in this tool. Namely - support for retention days/weeks/months.

Since you mention that this tool is based on **postgres-backup-s3**, why don't you just let it support uploading to S3?

I've asked for the same in **postgres-backup-s3** - https://github.com/eeshugerman/postgres-backup-s3/issues/40

As for **rclone**, it's not convenient to use for this one purpose and would be an extra tool to maintain, especially due to lack of scheduling support in its docker implementation.

Thanks!

**Reply by buncis** [reply]
have u check this? https://github.com/tiredofit/docker-db-backup
